src/dmd/dtoh.d

```diff
@@ -0,0 +1,1496 @@
+        "dcast.d", "dclass.d", "declaration.d", "delegatize.d", "denum.d", "dimport.d", "dinterpret.d",
+        "dmacro.d", "dmangle.d", "dmodule.d", "doc.d", "dscope.d", "dstruct.d", "dsymbol.d", "dsymbolsem.d", "dtemplate.d", "dversion.d",
+        "entity.d", "errors.d", "escape.d", "expression.d", "expressionsem.d",
+        "func.d",
+        "globals.d", "gluelayer.d",
+        "hdrgen.d",
+        "iasm.d", "id.d", "identifier.d", "impcnvtab.d", "imphint.d", "init.d", "initsem.d", "inline.d", "inlinecost.d", "intrange.d",
+        "json.d",
+        "lambdacomp.d", "lexer.d",
+        "mtype.d",
+        "nogc.d", "nspace.d",
+        "objc.d", "opover.d", "optimize.d",
+        "parse.d", "printast.d",
+        "safe.d", "sapply.d", "semantic2.d", "semantic3.d", "sideeffect.d", "statement.d",
+        "statementsem.d", "staticassert.d", "staticcond.d",
+        "target.d", "templateparamsem.d", "tokens.d", "traits.d", "typesem.d", "typinf.d",
+        "utf.d", "utils.d",
+    ];
+}
+
+int main(string[] args)
+{
+    import std.algorithm.sorting : sort;
+    import std.array : array;
+    import std.file : readText;
+    import std.path : baseName, buildPath, dirName;
+    import std.string : toStringz;
+
+    import dmd.dsymbolsem;
+    import dmd.errors;
+    import dmd.id;
+    import dmd.dinifile;
+    import dmd.parse;
+
+    import dmd.semantic2;
+    import dmd.semantic3;
+    import dmd.builtin : builtin_init;
+    import dmd.dmodule : Module;
+    import dmd.expression : Expression;
+    import dmd.frontend;
+    import dmd.objc : Objc;
+    import dmd.root.response;
+    import dmd.root.stringtable;
+    import dmd.target : Target;
+    
+    import core.memory;
+    import core.stdc.stdio : printf;
+    
+    GC.disable();
+    initDMD();
+    
+    Strings arguments = Strings(args.length);
+    for (size_t i = 0; i < args.length; i++)
+    {
+        arguments[i] = args[i].ptr;
+    }
+    if (response_expand(&arguments)) // expand response files
+        error(Loc.initial, "can't open response file");
+    auto files = Strings(arguments.dim - 1);
+    global.params.argv0 = args[0];
+
+    
+    global.inifilename = parse_conf_arg(&arguments);
+    if (global.inifilename)
+    {
+        // can be empty as in -conf=
+        if (strlen(global.inifilename) && !FileName.exists(global.inifilename))
+        error(Loc.initial, "Config file '%s' does not exist.", global.inifilename);
+    }
+    else
+    {
+        version (Windows)
+        {
+            global.inifilename = findConfFile(global.params.argv0, "sc.ini").ptr;
+        }
+        else version (Posix)
+        {
+            global.inifilename = findConfFile(global.params.argv0, "dmd.conf").ptr;
+        }
+        else
+        {
+            static assert(0, "fix this");
+        }
+    }
+    // Read the configurarion file
+    auto inifile = File(global.inifilename);
+    inifile.read();
```

Comment:
I think most of this code in this `main` method could be saved by using DMD's `dmd.frontend`: 
- https://github.com/dlang/dmd/blob/master/src/dmd/frontend.d
- https://github.com/dlang/dmd/blob/master/test/dub_package/frontend.d